### ISSUE TITLE
sort-comp: `static defaultProps` must be placed under componentWillUnmount()

### ISSUE THREAD

**Issue by AndrewRevinsky** [yannickcr/eslint-plugin-react#122]
For some reason, the ES6 class composition for React isn't taken into account. 

For all obvious reasons, while this is supposed to be the right sort order..
```
class AccountInfo extends Component {

  static propTypes = {
    accountId: PropTypes.string.isRequired
  };

  static defaultProps = {
    accountId: ''
  };

```
..a warning shows that says:
`warning  defaultProps must be placed after componentWillUnmount  react/sort-comp`

**Comment by mathieumg** [comment]
I'm curious, if you manually add `defaultProps` after `propTypes` in the lifecycle configuration, does it solve it? If so, it could be added at the same time as `constructor` is, see: https://github.com/yannickcr/eslint-plugin-react/issues/97#issuecomment-110283110

**Comment by AndrewRevinsky** [comment]
I edited `.eslitrc` to contain this line:

```json
    "react/sort-comp": [1, {
      "order": [
        "constructor",
        "lifecycle",
        "everything-else",
        "render"
      ],
```

Still, I get this:
`28:2  warning  defaultProps must be placed after componentWillUnmount  react/sort-comp`
for the ..

```jsx
class AccountInfo extends Component {

  static propTypes = {
    accountId: PropTypes.string.isRequired
  };

  //noinspection Eslint
  static defaultProps = {
    accountId: ''
  };
```

**Comment by AndrewRevinsky** [comment]
Oh, no. Now I get an error, actually:
```
 throw e;
                      ^
YAMLException: Cannot read config file: [...]\.eslintrc
Error: JS-YAML: missed comma between flow collection entries at line 57, column 1:
    }
    ^

Process finished with exit code 1
```

**Comment by AndrewRevinsky** [comment]
And.. I am still getting this error regarding the constuctor placement:
```
CreateAccount.js
   36:2   warning  constructor must be placed after componentWillUnmount  react/sort-comp
  134:16  warning  Empty components are self-closing                      react/self-closing-comp
  139:16  warning  Empty components are self-closing                      react/self-closing-comp
```

**Comment by yannickcr** [comment]
ES6 methods are not set in the default `lifecycle` configuration (yet). This should be fixed when I'll work on #97 

So, for now you can set up a custom `lifecycle` configuration in your .eslintrc:

```json
...
"react/sort-comp": [1, {
  "groups": {
    "lifecyle": [
      "constructor",
      "displayName",
      "propTypes",
      "contextTypes",
      "childContextTypes",
      "statics",
      "defaultProps",
      "childContext",
      "componentWillMount",
      "componentDidMount",
      "componentWillReceiveProps",
      "shouldComponentUpdate",
      "componentWillUpdate",
      "componentDidUpdate",
      "componentWillUnmount"
    ]
  }
}]
...
```

You can find more informations about this rule configuration settings in [the documentation](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md).
